### ISSUE TITLE
[CI][ROCm] inductor-periodic gfx950 dynamo benchmarks red since ROCm 7.2 -> 7.14 upgrade (#188429)

### ISSUE THREAD

**Issue by bobrenjc93** [pytorch/pytorch#192862]
TITLE: [CI][ROCm] inductor-periodic gfx950 dynamo benchmarks red since ROCm 7.2 -> 7.14 upgrade (#188429)

LABELS: module: rocm, module: inductor, oncall: pt2, module: ci, high priority

BODY:

`inductor-periodic` has been red on main continuously since 2026-08-08, on the
`rocm-periodic-dynamo-benchmarks-test` (gfx950) jobs. No non-ROCm job in the
workflow is failing. Because the workflow never goes green, it is currently not
able to surface any real inductor regression.

## Root cause

The last green `inductor-periodic` run was `0fc30948e35` (2026-08-07 20:58 UTC).
The first run of the continuous red streak was `7c898a118f0` (2026-08-08 05:03 UTC).

`bb96371605b` -- "[ROCm][CI] Switch rocm-preview to TheRock wheels and upgrade
CI to 7.14 (#188429)" -- landed at 2026-08-07 22:20 UTC, inside that window.

Although the PR title scopes it to `rocm-preview`, it modified
`.ci/docker/common/install_rocm.sh` (+40/-270) and `.ci/docker/ubuntu-rocm/Dockerfile`.
Those are shared image definitions, so the content hash of the
`pytorch-linux-jammy-rocm-n-py3-benchmarks` image changed and *every* ROCm job
picked up the new image, including this periodic benchmark job.

Confirmed by diffing the two runs directly:

| | last green (Aug 7 20:58) | first red (Aug 8 05:03) |
|---|---|---|
| image tag | `...-benchmarks-d233d238439a...` | `...-benchmarks-0c3a9da87a44...` |
| `BUILD_ROCM_VERSION` | `7.2` | `7.14` |
| ROCm root | `/opt/rocm` | `site-packages/_rocm_sdk_devel` (TheRock wheel) |

So this is a ROCm 7.2 -> 7.14 runtime jump on gfx950, not an inductor change.

## Symptoms

Three distinct model/suite failures, stable across 9 consecutive runs
(Aug 8 05:03 through Aug 10 17:02):

1. `torch_multimodal_clip` -- `dynamic_inductor_torchbench` shard 2,
   `accuracy=fail_accuracy, expected=pass`. **9/9 runs, 100% consistent.**
   This is the one that behaves like a genuine numerical change under 7.14.

2. `vit_base_patch14_dinov2.lvd142m` -- `dynamic_inductor_timm` and
   `aot_inductor_timm` shard 2 (inference, `expected=pass`),
   `accuracy=eager_two_runs_differ`. 8/9 runs. Note `eager_two_runs_differ`
   means two *eager* runs disagree -- inductor is not involved at all, the
   model is nondeterministic on this hardware/runtime.

3. `convnextv2_nano.fcmae_ft_in22k_in1k` -- `dynamic_inductor_timm` shard 1
   (training, `expected=fail_accuracy`), reports
   `IMPROVED: accuracy=pass`. 8/9 runs. `check_accuracy.py` exits nonzero on
   improvements too, so a stale pessimistic baseline keeps the job red.
   Overlaps with #189893.

Note: the `EAGER_FAILED` lines in these logs come from `check_graph_breaks.py`
and appear for every model including passing ones. They are not part of the
failure; only `check_accuracy.py` sets the nonzero exit.

## Suggested next steps

- (2) and (3) are baseline/nondeterminism problems and should be rebaselined
  under 7.14 via `benchmarks/dynamo/ci_expected_accuracy/update_expected.py`.
- (1) should be confirmed as an accepted numerical change under ROCm 7.14
  rather than blanket-rebaselined, since it is 100% reproducible and one-directional.
- Longer term, an image-definition change that rebuilds the shared ROCm image
  should probably run the ROCm periodic benchmarks before landing.

cc @ezyang @gchanan @kadeng @msaroufim @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @jataylo @hongxiayang @naromero77amd @pragupta @jerrymannil @xinyazhang @malfet @pytorch/pytorch-dev-infra @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @muchulee8 @amjames @aakhundov @coconutruben @ethanwee1

**Comment by bobrenjc93** [comment]
Escalating: this is now **14 consecutive failed `inductor-periodic` runs**, going
back to 2026-08-08. The workflow has not been green for 4 days, so it is
currently incapable of catching any real inductor regression - which is the
actual cost here, beyond the ROCm failures themselves.

Still unassigned after ~18 hours, so tagging people directly.

Restating the root cause from the description, since it is the actionable part:
the failures start in the exact window where `bb96371605b`
("[ROCm][CI] Switch rocm-preview to TheRock wheels and upgrade CI to 7.14",
#188429) landed. Although that PR is scoped to `rocm-preview` by title, it
modified `.ci/docker/common/install_rocm.sh` and `.ci/docker/ubuntu-rocm/Dockerfile`,
which are shared image definitions - so the `pytorch-linux-jammy-rocm-n-py3-benchmarks`
image was rebuilt and *every* ROCm job picked up the new stack, including this one:

|  | last green (Aug 7 20:58) | first red (Aug 8 05:03) |
|---|---|---|
| image tag | `...-benchmarks-d233d238439a...` | `...-benchmarks-0c3a9da87a44...` |
| `BUILD_ROCM_VERSION` | `7.2` | `7.14` |
| ROCm root | `/opt/rocm` | `site-packages/_rocm_sdk_devel` (TheRock wheel) |

Suggested split of the work, since these are not all the same problem:

- `vit_base_patch14_dinov2.lvd142m` reports `eager_two_runs_differ`, i.e. two
  *eager* runs disagree. Inductor is not involved; this is model
  nondeterminism under 7.14 and wants a rebaseline or a flaky-model entry.
- `convnextv2_nano.fcmae_ft_in22k_in1k` is pinned to `fail_accuracy` but now
  passes, so `check_accuracy.py` fails the job on the *improvement*. Stale
  baseline, overlaps #189893.
- `torch_multimodal_clip` is the one I would not rebaseline blindly - it fails
  100% consistently and one-directionally, which looks like a genuine numerical
  change under 7.14 rather than noise.

If a rebaseline under 7.14 is the agreed path for the first two, happy to put
that up; I mainly want confirmation on the third before anyone papers over it.

cc @jeffdaily @jataylo @naromero77amd @ethanwee1

**Comment by naromero77amd** [comment]
@bobrenjc93 Thanks for reporting it. I am happy to take you up on the offer for a PR to rebaseline them. We will go back and triage later. 

For the first two, let's rebase with `eager_two_runs_differ` and `pass`, respectively.

For the third one -- yes, it looks suspicious. Let's do `fail_accuracy` for now.

Let's keep this issue open so we can keep track of it.

**Comment by bobrenjc93** [comment]
@naromero77amd I put up #193216 with the values you specified, ran it against
the real gfx950 jobs, and **the pins do not hold**. Reporting back before
landing since this contradicts the plan we agreed.

Two of the four jobs went green; two stayed red, both the same way:

```
dynamic_inductor_timm 2/2        vit_base_patch14_dinov2  IMPROVED: accuracy=pass, expected=fail_accuracy
dynamic_inductor_torchbench 2/2  torch_multimodal_clip    IMPROVED: accuracy=pass, expected=fail_accuracy
```

Both of those are my errors, not yours -- you were working from what I told you:

- I said `torch_multimodal_clip` failed 100% consistently and one-directionally,
  which is why we agreed on `fail_accuracy`. It **passed** on the very first run
  after being pinned. My 9/9 sample was real but not predictive, so the "looks
  like a genuine numerical change" read is not supported.
- I changed `vit_base_patch14_dinov2` only on the inference baselines, since
  only those were failing, and flagged that as a deliberate deviation from
  "rebase vit with `eager_two_runs_differ`". The untouched **training** baseline
  then went red on an improvement.

The general problem is that `check_accuracy.py` fails a job on a mismatch in
either direction, so for a model that genuinely flips, any pinned value just
selects which outcome turns CI red.

I have switched the PR to add all three to the ROCm-scoped `flaky_models` set
instead, which yields `PASS_BUT_FLAKY`/`FAIL_BUT_FLAKY` and appends to neither
`failed` nor `improved`. That block already carries `alexnet` and `demucs` under
the comment *"Discovered on gfx950 CI after ROCm 7.2 upgrade, eager mode non
determinism"*, which is the same root cause. Simulating every observed outcome
through the real `check_accuracy()`:

```
observed=pass                   job_fails=False
observed=eager_two_runs_differ  job_fails=False
observed=fail_accuracy          job_fails=False
```

Two things I want your call on, since this is your area:

1. This gives up accuracy signal for all three models on ROCm. That matches the
   `alexnet`/`demucs` precedent and your "we will go back and triage later", but
   it does mean a real regression in them would go unnoticed until someone
   revisits.
2. If you would rather keep real signal for `torch_multimodal_clip`, the
   narrower option is extending `get_eager_nondeterministic_models()` to cover
   the ROCm paths -- it currently returns entries only for two CUDA files. That
   tolerates only `pass -> eager_two_runs_differ` and still catches a genuine
   `fail_accuracy`. No existing ROCm precedent for it, but it preserves more
   signal.

Happy to go either way. Keeping this issue open regardless, as you asked.

**Comment by bobrenjc93** [comment]
CI update: the latest inductor-periodic branch run for #193216 finished with every ROCm/gfx950 job green under the current flaky_models change: https://github.com/pytorch/pytorch/actions/runs/31715044756

The only failing job in that run was dynamic_cpu_inductor_timm 2/2: https://github.com/pytorch/pytorch/actions/runs/31715044756/job/94501551729. Its logs show an unrelated offline Hugging Face cache failure for several timm models, not a failure in the ROCm coverage or this change.

This validates the current change against the incident. @naromero77amd, could you confirm whether you prefer the current ROCm-scoped flaky_models approach or the narrower eager-nondeterministic handling described above? I am keeping the PR in draft pending that call.

**Comment by naromero77amd** [comment]
@bobrenjc93 Thanks for your help again and the detailed analysis. Let's go with the ROCm-scoped flaky_models approach.

**Comment by bobrenjc93** [comment]
Thanks, @naromero77amd. I’ve recorded the ROCm-scoped `flaky_models` decision in #193216 and moved the PR out of draft. The final diff only changes the ROCm flaky-model set, and the gfx950 validation run has all ROCm jobs green. Review is already requested from you, @jeffdaily, and @jataylo.

I’ll leave this issue open, as requested, to track the underlying nondeterminism and eventual removal of the exemptions.
